Automattic/wp-calypso
# Notifications: add keyboard shortcut info

Fixes https://github.com/Automattic/wp-calypso/issues/91358

## Proposed Changes

* Adds a Keyboard shortcuts info popup to the notification panel 
* Keyboard shortcut popover can be opened by either clicking on the info icon or by using the keyboard shortcut `i`.
* Uses the Popover component from the `@automattic/components`
* Hidden when within mobile user agents detected.

## Why are these changes being made?


* Users aren't aware of what keyboard shortcuts are available 

## Testing Instructions



* Go to `calypso.live/home/{test-site}`
* Click on notifications icon and you should see

<img width="419" alt="Screenshot 2024-06-24 at 16 33 49" src="https://github.com/Automattic/wp-calypso/assets/5560595/85583752-151d-4c91-90bc-f429b352a233">

* Confirm popup works as expected by clicking the the info icon or using the `i` key.

* Run `install-plugin.sh notifications add/notifications-shortcut-info` to install changes on sandbox
* Sandbox a test site
* Go to that test site and confirm the notifications panel loads with the keyboard shortcuts icon/popup
* Should looks the same as on Calypso;

<img width="417" alt="Screenshot 2024-06-24 at 16 34 56" src="https://github.com/Automattic/wp-calypso/assets/5560595/e6bb3c9c-2d4f-443c-ac06-054b8907b51f">

* Go to Reader notifications and confirm looks ok;

<img width="711" alt="Screenshot 2024-06-24 at 16 35 13" src="https://github.com/Automattic/wp-calypso/assets/5560595/672ed60a-07f1-4194-9275-3a847a16e303">

* Confirm you can't see the info icon when in mobile views

<img width="537" alt="Screenshot 2024-06-25 at 16 50 01" src="https://github.com/Automattic/wp-calypso/assets/5560595/c91f981a-cde2-47d6-8d02-1fe8b8d3ff0d">

## Pre-merge Checklist
